src/python/pants/backend/go/pkg.py

```diff
@@ -29,8 +36,8 @@
 # TODO: Add class docstring with info on the fields.
 @dataclass(frozen=True)
 class ResolvedGoPackage:
-    # Address of the `go_package` target. Copied from the `ResolveGoPackageRequest` for ease of access.
-    address: Address
+    # Address of the `go_package` target (if any).
+    address: Optional[Address]
```

Comment:
Not blocking, but FYI you can do this

```suggestion
    address: Address | None
```

ditto the fields below

Reply:
Done (for all applicable types in the file).